snuba/datasets/storages/factory.py

```diff
@@ -33,15 +34,27 @@ def __init__(self) -> None:
             self.__initialize()
 
     def __initialize(self) -> None:
-        self._config_built_storages = {
-            storage.get_storage_key(): storage
-            for storage in [
-                build_storage_from_config(config_file)
-                for config_file in glob(
-                    settings.STORAGE_CONFIG_FILES_GLOB, recursive=True
-                )
-            ]
-        }
+        if get_config("load_configs_multi_processed", 0):
+            with Pool() as p:
+                self._config_built_storages = {
+                    storage.get_storage_key(): storage
+                    for storage in p.map(
+                        build_storage_from_config,
+                        glob(settings.STORAGE_CONFIG_FILES_GLOB, recursive=True),
+                    )
+                }
+            for key in self._config_built_storages:
+                register_storage_key(key.value)
```

Comment:
Need to reregister the keys since multiprocessing only returns the built storage object and creates a copy of the `StorageKey` enum within the process which the key gets registered to, leaving the StorageKey enum in the main process unaltered.